packages/ra-core/src/sideEffect/notification.js

```diff
@@ -0,0 +1,22 @@
+import { put, takeEvery } from 'redux-saga/effects';
+import { showNotification } from '../actions/notificationActions';
+
+/**
+ * Notification Side Effects
+ */
+function* handleNotification({ type, meta: { notification } }) {
+    const { body, level = 'info', messageArgs = {} } = notification;
+    yield put(
+        showNotification(body, level, {
+            messageArgs,
+            undoable: type.indexOf('_OPTIMISTIC') !== -1,
```

Comment:
Not fan of the optimistic case detection mechanism. Can't we use a `optimistic` property in the metas instead ?